src/openrct2/management/NewsItem.cpp

```diff
@@ -295,6 +295,20 @@ std::optional<CoordsXYZ> news_item_get_subject_location(int32_t type, int32_t su
     return subjectLoc;
 }
 
+static NewsItem* gNewsItems_first_open_slot()
+{
+    NewsItem* newsItem = gNewsItems;
+
+    while (newsItem->Type != NEWS_ITEM_NULL)
```

Comment:
Can we change this into a more normal loop like a for loop or even better use some std functions like a std::find

Reply:
The issue is this function does two things: find the first open spot and close items. The "iterator" is not incremented on every iteration because a new spot may open by closing the current item.

To transform this in a regular loop we'd need to separate the two concerns: closing items and finding the first open spot.

Reply:
Splitting it into two would make more sense i think.